src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeed.cs

```diff
@@ -0,0 +1,249 @@
+            {
+                using (var streamWriter = new StreamWriter(fs))
+                {
+                    using (JsonTextWriter writer = new JsonTextWriter(streamWriter))
+                    {
+                        writer.WriteStartObject();
+                        writer.WritePropertyName("versions");
+                        writer.WriteStartArray();
+                        writer.WriteRawValue($"\"{value}\"");
+                        writer.WriteEndArray();
+                        writer.WriteEndObject();
+                    }
+                }
+            }
+        }
+
+        private void GenerateRootServiceIndex(string accountName, string redirectUrl)
+        {
+            string pathToTempFolder = Path.Combine(Directory.GetCurrentDirectory(), "tmp", redirectUrl);
+            if (Directory.Exists(pathToTempFolder))
+            {
+                Directory.Delete(pathToTempFolder, true);
+            }
+            Directory.CreateDirectory(pathToTempFolder);
+            using (FileStream fs = File.OpenWrite(Path.Combine(pathToTempFolder, "index.json")))
+            {
+                using (var streamWriter = new StreamWriter(fs))
+                {
+                    using (JsonTextWriter writer = new JsonTextWriter(streamWriter))
+                    {
+                        writer.WriteStartObject();
+                        writer.WritePropertyName("version");
+                        writer.WriteRawValue("\"3.0.0\"");
+                        writer.WritePropertyName("resources");
+                        writer.WriteStartArray();
+                        writer.WriteStartObject();
+                        writer.WritePropertyName("@id");
+                        writer.WriteRawValue($"\"{ GetFeedUrl }{ContainerName}/{redirectUrl}/\"");
+                        writer.WritePropertyName("@type");
+                        writer.WriteRawValue("\"PackageBaseAddress/3.0.0\"");
+                        writer.WritePropertyName("comment");
+                        writer.WriteRawValue("\"Base URL of Azure storage where NuGet package registration info for intermediaries is stored.\"");
+                        writer.WriteEndObject();
+                        writer.WriteEndArray();
+                        writer.WritePropertyName("@context");
+                        writer.WriteStartObject();
+                        writer.WritePropertyName("@vocab");
+                        writer.WriteRawValue("\"http://schema.nuget.org/schema#\"");
+                        writer.WritePropertyName("comment");
+                        writer.WriteRawValue("\"http://www.w3.org/2000/01/rdf-schema#comment\"");
+                        writer.WriteEndObject();
+                        writer.WriteEndObject();
+                    }
+                }
+            }
+        }
+
+        public bool IsSanityChecked(IEnumerable<string> items)
```

Comment:
I don't see any users of this? What is its purpose exactly?

Reply:
I wanted a quick check for files that aren't nupkgs and are duplicated. Will invoke in the right place.